Ilya/NetflixRoulette/netflix-roulette/src/shared/searchMovie/SearchMovie.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useEffect, useState } from "react";
+import { useHistory, useLocation } from 'react-router-dom';
+import queryString from "query-string";
+
+import "./SearchMovie.scss";
+import SearchBtn from "shared/button/SearchBtn";
+
+interface ISearchMovieProps {
+    onSubmit(search: string, searchBy: string): void,
+}
+
+export enum SearchBy {
+    Genre = "genres",
+    Title = "title"
+}
+
+function SearchMovie(props: ISearchMovieProps) {
+
+    const [value, setValue] = useState("");
+    const [searchBy, setSearchBy] = useState(SearchBy.Title as string);
+
+    let history = useHistory();
+    let location = useLocation();
+
+    useEffect(() => {
+        const parsed = queryString.parse(location.search);
+        setValue(parsed.search as string || "");
+        setSearchBy(parsed.searchBy as string || SearchBy.Title);
+    }, [location.search]);
```

Comment:
You can remove this useEffect. The useEffect that you have below will be called on mount.